File: 0946-validate-stack-sequences/0946-validate-stack-sequences.py
class Solution:
    def validateStackSequences(self, pushed: List[int], popped: List[int]) -> bool:
        i, stack = 0, []
        for p in pushed:
            stack.append(p)
            while stack and i < len(popped) and stack[-1] == popped[i]:
                stack.pop()
                i += 1
        return i == len(popped)

File: 0946-validate-stack-sequences/0946-validate-stack-sequences.js
/**
 * @param {number[]} pushed
 * @param {number[]} popped
 * @return {boolean}
 */
var validateStackSequences = function(pushed, popped) {
    let i = 0
    const stack = []
    for (const p of pushed) {
        stack.push(p)
        while (stack.length && i < popped.length && stack[stack.length-1] === popped[i]) {
            stack.pop()
            i++
        }
    }
    return i === popped.length
};